### ISSUE TITLE
[Web] SelectionArea Doesn't Respect Text's TextAlign in Web Platfrom

### ISSUE THREAD

**Issue by Ali-Fadaei** [flutter/flutter#152435]
### Steps to reproduce

run this code with both web and win/android platforms:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Selection Area Text Align Issue',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const Scaffold(
        body: Padding(
          padding: EdgeInsets.symmetric(horizontal: 16),
          child: Center(
            child: Row(
              children: [
                Expanded(
                  child: SelectionArea(
                    child: Text(
                      'text inside Selection Area with textAlign: TextAlign.start',
                      textAlign: TextAlign.start,
                    ),
                  ),
                )
              ],
            ),
          ),
        ),
      ),
    );
  }
}
```

### Expected results

![Screenshot 2024-07-27 163832](https://github.com/user-attachments/assets/05833c51-61bb-43ad-bf34-d17246715f1f)

### Actual results

![Screenshot 2024-07-27 164116](https://github.com/user-attachments/assets/bc36ce85-f622-4199-82f9-d3e3fb568e2d)

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Selection Area Text Align Issue',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const Scaffold(
        body: Padding(
          padding: EdgeInsets.symmetric(horizontal: 16),
          child: Center(
            child: Row(
              children: [
                Expanded(
                  child: SelectionArea(
                    child: Text(
                      'text inside Selection Area with textAlign: TextAlign.start',
                      textAlign: TextAlign.start,
                    ),
                  ),
                )
              ],
            ),
          ),
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.22.2, on Microsoft Windows [Version 10.0.22621.3880], locale en-US)
    • Flutter version 3.22.2 on channel stable at C:\flutter\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 761747bfc5 (7 weeks ago), 2024-06-05 22:15:13 +0200
    • Engine revision edd8546116
    • Dart version 3.4.3
    • DevTools version 2.34.3

[✓] Windows Version (Installed version of Windows is version 10 or higher)

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\Alifa\AppData\Local\Android\sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11572160)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.10.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.10.35013.160
    • Windows 10 SDK version 10.0.22621.0

[✓] Android Studio (version 2023.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11572160)

[✓] VS Code (version 1.91.1)
    • VS Code at C:\Users\Alifa\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.92.0

[✓] Connected device (2 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.3880]
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 127.0.2651.74
```

</details>

**Comment by jaypal1046** [comment]
Hi @Ali-Fadaei ,

Please remove the Expanded widget from the code. The issue is happening because Expanded is giving the text full width, causing the text to be centered with space on both sides.

**Comment by darshankawar** [comment]
@Ali-Fadaei 
Check above and see if it helps to resolve your case or not.

**Comment by Ali-Fadaei** [comment]
> Hi @Ali-Fadaei ,
> 
> Please remove the Expanded widget from the code. The issue is happening because Expanded is giving the text full width, causing the text to be centered with space on both sides.

this code sample works fine on android + windows platforms. its just happened in web!
I use AutoSizeText For Real App, and its should wrap with expanded inside a row. 
Also i can fix this Issue by wrapping the SelectionArea with Align widget inside Expanded. but the point is SelectionArea should Respect TextAlign in web, same as other platforms. isn't it?

**Comment by darshankawar** [comment]
This look similar to https://github.com/flutter/flutter/issues/121053 which has been fixed that you can check and confirm.

**Comment by Ali-Fadaei** [comment]
@darshankawar 
yes both are similar, and it has a merged fix PR, if the PR cherrypicked to stable, the issue already exists.

**Comment by darshankawar** [comment]
I don't see that fix making it to 3.22.0 stable looking at [release notes](https://docs.flutter.dev/release/release-notes/release-notes-3.22.0), so it is likely that it is not in stable yet, and maybe available in next stable release. You can check the same by running your scenario in latest master to confirm.
Closing this as already fixed.

**Comment by Ali-Fadaei** [comment]
@darshankawar 
thanks.
